src/mol-io/reader/lammps/traj/parser.ts

```diff
@@ -105,63 +105,127 @@ async function handleAtoms(state: State, count: number, parts: string[]): Promis
-            // Periodicity of the box
+        } else if (trimmedLine.includes('ITEM: BOX BOUNDS')) {
+            // Process box bounds
+            const tokens = trimmedLine.split('ITEM: BOX BOUNDS ')[1].split(' ');
             const px = tokens[0];
             const py = tokens[1];
             const pz = tokens[2];
-            // the actual box bounds
-            const xbound = readLine(state.tokenizer).trim().split(' ');
-            const ybound = readLine(state.tokenizer).trim().split(' ');
-            const zbound = readLine(state.tokenizer).trim().split(' ');
+
+            const [xboundLine, xNewIndex] = readNextLine(data, currentIndex, decoder);
+            currentIndex = xNewIndex;
+            const xbound = xboundLine.trim().split(' ');
+
+            const [yboundLine, yNewIndex] = readNextLine(data, currentIndex, decoder);
+            currentIndex = yNewIndex;
+            const ybound = yboundLine.trim().split(' ');
+
+            const [zboundLine, zNewIndex] = readNextLine(data, currentIndex, decoder);
+            currentIndex = zNewIndex;
+            const zbound = zboundLine.trim().split(' ');
+
             const xlo = parseFloat(xbound[0]);
             const xhi = parseFloat(xbound[1]);
             const ylo = parseFloat(ybound[0]);
             const yhi = parseFloat(ybound[1]);
             const zlo = parseFloat(zbound[0]);
             const zhi = parseFloat(zbound[1]);
+
             f.bounds.push({
                 lower: [xlo, ylo, zlo],
                 length: [xhi - xlo, yhi - ylo, zhi - zlo],
-                periodicity: [px, py, pz] });
+                periodicity: [px, py, pz]
+            });
         }
     }
+
+    // Calculate time offsets
     if (f.times.length >= 1) {
         f.timeOffset = f.times[0];
     }
     if (f.times.length >= 2) {
         f.deltaTime = f.times[1] - f.times[0];
     }
+
     return Result.success(f);
 }
 
-export function parseLammpsTrajectory(data: string) {
+/**
+ * Reads the next line from a Uint8Array starting from a specific index.
+ * It decodes and returns the next line and the updated index for the next call.
+ */
+function readNextLine(
+    data: Uint8Array,
+    currentIndex: number,
+    decoder: TextDecoder
+): [line: string, nextIndex: number] {
+    let line = '';
+    let nextIndex = currentIndex;
+
+    for (let i = currentIndex; i < data.length; i++) {
+        if (data[i] === 10) { // ASCII code for '\n'
+            line = decoder.decode(data.subarray(currentIndex, i), { stream: true });
+            nextIndex = i + 1; // Update the next index to the position after '\n'
+            break;
+        }
+    }
+
+    // If no newline is found, decode the remaining part
+    if (nextIndex === currentIndex) {
+        line = decoder.decode(data.subarray(currentIndex + 1), { stream: false });
+        nextIndex = data.length; // Set the index to the end of the data
+    }
+
+    return [line.trim(), nextIndex];
```

Comment:
Given how big the files can be, this allocation and the subsequent deconstructions are very expensive. Instead of this I would suggest passing a "result array (or object)" as an argument and adding to it here.

Also, you are calling the trim twice.